app/assets/javascripts/components/articles/article_graphs.jsx

```diff
@@ -1,170 +1,148 @@
-import PropTypes from 'prop-types';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import Wp10Graph from './wp10_graph.jsx';
 import EditSizeGraph from './edit_size_graph.jsx';
 import Loading from '../common/loading.jsx';
-import request from '../../utils/request';
+import request from '../../utils/request.js';
 
+const ArticleGraphs = ({ article }) => {
+  const { id: article_id } = article;
 
-const ArticleGraphs = createReactClass({
-  displayName: 'ArticleGraphs',
+  const [showGraph, setShowGraph] = useState(false);
+  const [selectedRadio, setSelectedRadio] = useState('wp10_score');
+  const [articleData, setArticleData] = useState(null);
 
-  propTypes: {
-    article: PropTypes.object
-  },
+  const elementRef = useRef(null);
 
-  getInitialState() {
-    return {
-      showGraph: false,
-      selectedRadio: 'wp10_score',
-      articleData: null
-    };
-  },
-
-  componentDidMount() {
-    this.ref = React.createRef();
-  },
-
-  componentDidUpdate(_, prevState) {
-    if (this.state.showGraph && !prevState.showGraph) {
-      // Add event listener when the component is visible
-      document.addEventListener('mousedown', this.handleClickOutside);
+  useEffect(() => {
+    if (showGraph) {
+      document.addEventListener('mousedown', handleClickOutside);
+    } else {
+      document.removeEventListener('mousedown', handleClickOutside);
     }
-    if (!this.state.showGraph && prevState.showGraph) {
-      // remove event listener when the component is hidden
-      document.removeEventListener('mousedown', this.handleClickOutside);
-    }
-  },
 
-  getData() {
-    if (this.state.articleData) { return; }
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [showGraph]);
 
-    const articleId = this.props.article.id;
-    const articledataUrl = `/articles/article_data.json?article_id=${articleId}`;
+  function getData() {
+    if (articleData) {
+      return;
+    }
+    const articledataUrl = `/articles/article_data.json?article_id=${article_id}`;
     request(articledataUrl)
       .then(resp => resp.json())
       .then((data) => {
-        this.setState({ articleData: data });
+        setArticleData(data);
       });
-  },
+  }
 
-  handleClickOutside(event) {
-    const element = this.ref.current;
+  function handleClickOutside(event) {
+    const element = elementRef.current;
     if (element && !element.contains(event.target)) {
-      this.hideGraph();
+      handleHideGraph();
+    }
+  }
+
+  function handleShowGraph() {
+    getData();
+    setShowGraph(true);
+  }
+
+  function handleHideGraph() {
+    setArticleData(null);
+    setShowGraph(false);
+  }
+
+  function handleRadioChange(event) {
+    setSelectedRadio(event.currentTarget.value);
+  }
+
+  function graphId() {
+    return `vega-graph-${article_id}`;
+  }
+
+  const dataIncludesWp10 = articleData && articleData[0].wp10;
+
+  const graph = useMemo(() => {
+    if (!articleData) {
```

Comment:
no need for the `useMemo`

Reply:
We need the use memo here as building the graph I based on the result gotten from calling getData, so the graph changes based on the result from calling getData. adding articleData as array dependency in the memo call allows the graph to be rendered with the new article data @TheTrio

Reply:
removing the useMemo still does that - it will get called every time state changes. Not sure I follow your logic here